src/Geometry/BoundingBox.h

```diff
@@ -1,43 +1,187 @@
 #pragma once
 
+// Local headers
 #include "Vector2.h"
 #include "Size.h"
 
+// Standard library headers
+#include <cmath>
+
 namespace Geometry {
-	template<class T>
-	struct BoundingBox {
-		Vector2<T> start;
-		Vector2<T> end;
-
-		T width() const;
-
-		T height() const;
-
-		Size<T> size() const;
-
-		bool contains(const Vector2<T> &point) const;
-	};
-
-	template<class T>
-	T BoundingBox<T>::width() const {
-		return end.x - start.x;
-	}
-
-	template<class T>
-	T BoundingBox<T>::height() const {
-		return end.y - start.y;
-	}
-
-	template<class T>
-	bool BoundingBox<T>::contains(const Vector2<T> &point) const {
-		return point.x >= start.x
-		       && point.x < end.x
-		       && point.y >= start.y
-		       && point.y < end.y;
-	}
-
-	template<class T>
-	Size<T> BoundingBox<T>::size() const {
-		return Size<T>(this->width(), this->height());
-	}
+  /// General class for representing a 2D maximum point
+  template<typename T>
+  class Maximum2 : public Vector2<T>
+  {
+  public:
+    /// Default constructor initialises nothing
+    Maximum2() = default;
```

Comment:
I would be tempted to not have a default constructor as that leaves us in an unknown state.  We should move to having classes always in a valid state (even if that is an explicit uninitialised state).

Reply:
Just uses the default constructor from Vector2<T> now.